core/meta/src/TCling.cxx

```diff
@@ -5283,8 +5283,8 @@ void TCling::UpdateClassInfoWithDecl(const void* vTD)
       }
       clang::QualType type( td->getTypeForDecl(), 0 );
 
-      auto declName=ND->getNameAsString().c_str();
-      if (!TClass::HasNoInfoOrEmuOrFwdDeclaredDecl(declName)){
+      auto declName = ND->getNameAsString();
```

Comment:
A string is constructed either way, the patch only extends lifetime of the object till the end of the function.